src/transformers/modeling_utils.py

```diff
@@ -1634,15 +1652,15 @@ def _autoset_attn_implementation(
                 and torch.cuda.device_count() > 1
             ):
                 logger.warning_once(
-                    "Using the `SDPA` attention implementation on multi-gpu setup with ROCM may lead to performance issues due to the FA backend. Disabling it to use alternative backends."
+                    "Using the `SDPA` attention implementation on multi-gpu setup with ROCM may lead to performance "
+                    "issues due to the FA backend. Disabling it to use alternative backends."
                 )
                 torch.backends.cuda.enable_flash_sdp(False)
         elif isinstance(requested_attn_implementation, dict):
             config._attn_implementation = None
         else:
             config._attn_implementation = "eager"
 
-        config._attn_implementation_autoset = True
```

Comment:
The actual change (3): when the user fully specifies the attention implementation, then this flag should be false. After these changes, it will be :)